completions/bash/clr-installer

```diff
@@ -0,0 +1,108 @@
+#!/bin/bash
+# shellcheck disable=SC2207
+# -----------------------------------------------------------------------
+#   Clear Linux OS* Installer - autocompletion script
+#
+#   Author: Lucius Hu - http://github.com/lebensterben
+#
+#   This program is free software: you can redistribute it and/or modify
+#   it under the terms of the GNU General Public License as published by
+#   the Free Software Foundation, version 2 or later of the License.
+#
+#   This program is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#   GNU General Public License for more details.
+#
+#   You should have received a copy of the GNU General Public License
+#   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+# -----------------------------------------------------------------------
+
+_clr_installer() {
+  local IFS opts prev cur
+
+  _init_completion -s || return
+
+  case "$prev" in
+    --archive|--copy-network|--iso|--keep-image|--reboot|--stub-image|--swupd-skip-diskspace-check|--telemetry)
```

Comment:
For consistency, we should include:
```suggestion
    --allow-insecure-http|--archive|--cfPurge|--copy-network|--copy-swupd|--high-contrast|--iso|--keep-image|--offline|--reboot|--stub-image|--swupd-clean|--swupd-skip-diskspace-check|--swupd-skip-optional|--telemetry|--tui)
```

Is there some standard to show what the default is for these true/false items? Most of these default to false, but a few default to true.

Reply:
The problem is I compiled the latest `clr-installer` but it was not in my PATH. So I actually missed some of those options.

And I should actually remove some of them.
So it would be
```bash
--archive|--copy-network|--copy-swupd|--keep-image|--reboot|--swupd-skip-diskspace-check)
```

If something defaults to true, you have to specify `--foo false` to turn it off.

But in the case of all other options that accepts a boolean, they defaults to false, and you just need `--foo` to turn it on.

P.S.
Also, it's rather verbose to have `--foo true|false`.
I suggests replace the 6 options that defaults to true by
- `--archive` -> `--no-archive`
- `--copy-network` -> `--no-copy-network`
- `--copy-swupd` -> `--no-copy-swupd`
- `--keep-image` -> `--no-keep-image`
- `--reboot` -> `--no-reboot`
- `--swupd-skip-diskspace-check` -> `--swupd-diskspace-check`

Reply:
For the booleans, the list should be what presented
`--allow-insecure-http|--archive|--cfPurge|--copy-network|--copy-swupd|--high-contrast|--iso|--keep-image|--offline|--reboot|--stub-image|--swupd-clean|--swupd-skip-diskspace-check|--swupd-skip-optional|--telemetry|--tui`

How are you determining the shorter list you are proposing?

I do like the idea of `no`, that is not a part of the standard flag parsing library we are using. Maybe sometime in the future we can upstream a fix.

Reply:
For the short list, well, I read the source code. I actually learnt Go
language just for writing completions for this last year.

Specifically, I checked `args/args.go`, for all `flag.BoolVar` and
`flag.BoolVarP`.

The followings defaults to true:
- `--reboot`
- `--swupd-skip-diskspace-check`
- `--archive`
- `--keep-image`
- `--copy-network`
And --copy-swupd` defaults to `true` for `interactive` as its description says.

My reasoning is:
- For mass installation with a config file, certainly it's avoids
  confusions to require the user to *explicitly* give the boolean value to
  those options.
- But when the installer is invoked from command line, it's quite rare to
  see patterns like `--foo=true`. Some GNU-style commands has
  `--(en|dis)able-FOO` or `--(no-|)FOO` options, that's a bit verbose but
  avoids any confusions at all.
- IMO, the problem of options accepting boolean arguments and defaults to
  `true` is:
  - Take `--swupd-skip-diskspace-check` for example, it defaults to `true`.
    When users see this, they may expect it to be `opt-in`. But it's
    actually the default behaviour. So they may writing explicitly
    `--swupd-skip-diskspace-check=true` when it's not necessary. Or they
    may not setting `--swupd-skip-diskspace-check=false` because they
    mis-understand how this option works.
  - This is especially an issue for `bash` users, because there's not way
    of showing descriptions in completion menu.
  - Surely users SHOULD read manual first before invoking the installer,
    but making careless mistakes is human nature....
- Therefore, I think that it's only necessary to provide that
  `true or false` choices for those defaults to `true`, to avoid any
  confusion. And for all that defaults to `false`, merely setting the flag
  without an argument is equivalent to setting it to `true`, which is
  concise and easy to understand.

Reply:
Okay, only offering the true or false option to booleans that default to true is fine.